test/integration/local/test_failure_scenarios.py

```diff
@@ -14,65 +14,107 @@
     """
     This test validates that all training processes finish before containers are shut down.
     """
     customer_script = 'all_processes_finish_customer_script.py'
-
-    cluster_size = 2
     hyperparameters = {'sagemaker_use_mpi': True, 'sagemaker_process_slots_per_host': 2,
                        'sagemaker_num_processes': 4}
 
-    local_mode.train(customer_script, resource_path, docker_image, opt_ml,
-                     hyperparameters=hyperparameters, source_dir=resource_path, use_gpu=use_gpu,
-                     cluster_size=cluster_size)
+    estimator = Chainer(entry_point=customer_script,
+                        source_dir=resource_path,
+                        role='SageMakerRole',
+                        image_name=docker_image,
+                        train_instance_count=2,
+                        train_instance_type='local',
+                        sagemaker_session=sagemaker_local_session,
+                        hyperparameters=hyperparameters,
+                        output_path='file://{}'.format(tmpdir))
 
-    file_name = 'output/data/process_could_complete'
+    estimator.fit()
 
-    assert local_mode.file_exists(opt_ml, file_name, host='algo-2'), 'Model was not saved'
+    completion_file = {'output': [os.path.join('data', 'algo-2', 'process_could_complete')]}
+    test_utils.files_exist(str(tmpdir), completion_file)
 
 
-def test_training_jobs_do_not_stall(docker_image, opt_ml, use_gpu):
+def test_training_jobs_do_not_stall(docker_image, sagemaker_local_session, tmpdir):
     """
     This test validates that training does not stall.
     https://github.com/chainer/chainermn/issues/236
     """
     customer_script = 'training_jobs_do_not_stall_customer_script.py'
-    cluster_size = 2
     hyperparameters = {'sagemaker_use_mpi': True, 'sagemaker_process_slots_per_host': 1,
                        'sagemaker_num_processes': 2}
 
-    local_mode.train(customer_script, resource_path, docker_image, opt_ml,
-                     hyperparameters=hyperparameters, source_dir=resource_path,
-                     use_gpu=use_gpu, cluster_size=cluster_size)
+    estimator = Chainer(entry_point=customer_script,
+                        source_dir=resource_path,
+                        role='SageMakerRole',
+                        image_name=docker_image,
+                        train_instance_count=2,
+                        train_instance_type='local',
+                        sagemaker_session=sagemaker_local_session,
+                        hyperparameters=hyperparameters,
+                        output_path='file://{}'.format(tmpdir))
+
+    # Local Mode doesn't export model/output artifacts upon failure
+    # https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/local/image.py#L133-L141
+    with pytest.raises(RuntimeError) as e:
+        estimator.fit()
 
-    assert local_mode.file_exists(opt_ml, 'output/failure'), 'Failure did not happen'
+    assert 'Failed to run:' in str(e)
```

Comment:
can we print out something more specific in the training script to assert here? the assertions here doesn't really guarantee that the job is failing the way we want it too. same with the other two tests

Reply:
talked offline - will fix the output export in the Python SDK first